packages/core/src/tools/utils.ts

```diff
@@ -660,8 +661,22 @@ export function removeDuplicates<T>(array: T[]) {
   return arrayFrom(set)
 }
 
-export function matchList(list: Array<string | RegExp>, value: string) {
-  return list.some((item) => item === value || (item instanceof RegExp && item.test(value)))
+export type MatchPredicate = (value: string) => boolean
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Use the 3 supported types:

```suggestion
export type MatchPredicate = string | RegExp | ((value: string) => boolean)
```
So you can simplify list types with `MatchPredicate[]`